ArchitectureFlow/ArchitectureFlow/Login/UseCase/LoginInteractor.h

```diff
@@ -0,0 +1,12 @@
+#import <Foundation/Foundation.h>
+#import "Login.h"
+
+@class LoginRequestToUserSessionMapper;
+@protocol EncryptionGateway;
+@protocol UserSessionGateway;
+
+@interface LoginInteractor : NSObject <Login>
```

Comment:
Do I get it right, that the protocol represents the usecase (though implicitly), the implementation is the interactor? Is there any practical advantage in having 2 names for the same thing?

Reply:
Transitioning from TID to NID can be a good example here: the same presenter is accessing different implementations for the same use case via a Login protocol.

Reply:
my answer above in another comment :)